contracts/zero-ex/contracts/src/features/multiplex/MultiplexOtc.sol

```diff
@@ -65,4 +65,47 @@ abstract contract MultiplexOtc is FixinEIP712 {
             state.boughtAmount = state.boughtAmount.safeAdd(makerTokenFilledAmount);
         } catch {}
     }
+
+    function _multiHopSellOtcOrder(
+        IMultiplexFeature.MultiHopSellState memory state,
+        IMultiplexFeature.MultiHopSellParams memory params,
+        bytes memory wrappedCallData
+    ) internal {
+        // Decode the tokens[], Otc order, and signature.
+        (address[] memory tokens, LibNativeOrder.OtcOrder memory order, LibSignature.Signature memory signature) = abi.decode(
+            wrappedCallData,
+            (address[], LibNativeOrder.OtcOrder, LibSignature.Signature)
+        );
+        // Validate tokens.
+        require(
+            tokens.length >= 2 &&
```

Comment:
Perhaps this is a dumb question: why allow `tokens` to have length `> 2`?

Reply:
this tokens array will contain all of the tokens in the swap, i.e. it needs to be more than 2 tokens to perform a multihop sell. we need token A, intermediate token, token B

Reply:
Your comment made me look into this in more detail. As far as I can tell, that's not enforced anywhere. `tokens` comes from the subcall data and isn't even signed. As best I can tell, it's totally unconstrained and this `require` is basically vacuous

Reply:
Bump

Reply:
Duncan is right, unless there is something I'm missing, you should be checking `params.tokens` against the `makerToken` and `takerToken` of the `OtcOrder`.

Something like
```
require(
    order.takerToken == params.tokens[state.hopIndex] &&
    order.makerToken == params.tokens[state.hopIndex + 1],
   "MultiplexOtcOrder::_multiHopSellOtcOrder/INVALID_TOKENS"
);
```

Reply:
Can we get rid of `tokens` altogether? Seems like it's not being used now